frontend/views/pages/Payments.vue

```diff
@@ -338,6 +405,48 @@ export default ({
     handleRowsPerPageChange (value) {
       this.ephemeral.rowsPerPage = value
       this.ephemeral.currentPage = 0 // go back to first page.
+    },
+    onRecordPaymentClick () {
+      this.$store.commit('addPaymentsToRecord', this.ephemeral.selectedTodoItems)
+      this.openModal('RecordPayment')
```

Comment:
Can you explain your reasoning a bit for why you decided to add these to the vuex state instead of passing them as parameters to the modal?

My understanding is that it's possible to pass parameters directly to the modal.

Reply:
Hey @taoeffect ,
That's good thing to know. 

I actually considering passing `hashes` of the payment items as the `queries` to the modal as the initial idea but it had two issues of
- url becoming way too long as the number of items increases
- having to re-organising the data to map them to the list item UI

So decided to use Vuex state and see if people agree with this way.
But as you pointed out above, I've found out that our `Modal.vue` component does have [`childData`](https://github.com/okTurtles/group-income/blob/2a4700348b46ecdd0423d75bd76b41960c69e443/frontend/views/components/modal/Modal.vue#L107) parameter to use for passing state directly to a modal.

So I have made changes ([commit](https://github.com/okTurtles/group-income/pull/1390/commits/df9098f0ce549f3b8c8510ac448116d069c9afa7)) that do things that way. pls let me know if any other issue.